### DISCUSSION TITLE
GraphQL or REST API query to retrieve the organization join date of all members in the organization

### DISCUSSION THREAD

**Discussion by sisaha9** [community/community#99057] [Apps, API and Webhooks]
Is there a GraphQL or REST API query to retrieve the organization join date of all members in the organization

**Comment by Naimulnobel** [comment]
For both GraphQL and REST, the ability to retrieve the organization join date of all members depends on the specific API you're using. 

For GitHub's GraphQL API, you can use the following query:

```graphql
{
  organization(login: "ORGANIZATION_NAME") {
    membersWithRole(first: 100) {
      edges {
        node {
          login
          createdAt
        }
      }
    }
  }
}
```

Replace `"ORGANIZATION_NAME"` with the name of your organization. This query retrieves the first 100 members of the organization along with their usernames and the date they created their accounts.

For GitHub's REST API, there isn't a direct way to get the join date of a user to an organization. You can only get the creation date of the user's account with the following endpoint:

`GET /users/:username`

This will return a JSON object with a `created_at` field, which is the date the user account was created.

Please note that you need to have the necessary permissions to access this information. Also, replace `:username` with the actual username.

**Reply by sisaha9** [reply]
So this gets the dates users created their accounts as you noted. Not when they joined the organization. I did find a way to get organization join dates by querying audit entries but it's only available in the Enterprise orgs

**Comment by naveen-98** [comment]
Hey!

For information regarding organization members, you might use the GitHub REST API `GET /orgs/:org/members` endpoint, which lists the members of the organization. The response provides details about each member, but it doesn't include the join date.

If you need the join date for each member, one possible approach is to retrieve the user details for each member individually using the GitHub REST API `GET /users/:username` endpoint. The response for a user contains information such as the creation date of the user account.

Here's a simplified example in Python using the `requests` library:

```python
import requests

def get_user_creation_date(username, access_token):
    headers = {
        "Authorization": f"Bearer {access_token}",
        "Accept": "application/vnd.github.v3+json",
    }

    response = requests.get(f"https://api.github.com/users/{username}", headers=headers)
    user_data = response.json()

    if "created_at" in user_data:
        return user_data["created_at"]
    else:
        return None

# Replace 'your_username', 'your_token' with your GitHub username and personal access token
user_creation_date = get_user_creation_date("your_username", "your_token")
print(f"User creation date: {user_creation_date}")
```

Please note that you need a GitHub personal access token with appropriate permissions to use these APIs.

**Reply by sisaha9** [reply]
So this gets the dates users created their accounts as you noted. Not when they joined the organization. I did find a way to get organization join dates by querying audit entries but it's only available in the Enterprise orgs
